src/Network_Equivalences-Communication.thy

```diff
@@ -91,6 +675,40 @@ lemma adapted_after_duploss:
   shows "\<currency>\<^sup>* A \<guillemotleft> \<E> = \<currency>\<^sup>* (A \<guillemotleft> \<E>)"
   by (simp only: duploss_def adapted_after_parallel adapted_after_loss adapted_after_duplication)
 
+lemma duploss_losing_transition:
+  shows "\<currency>\<^sup>* A \<rightarrow>\<^sub>s\<lparr>A \<triangleright> \<star>\<^bsup>n\<^esup> X\<rparr> (\<zero> \<parallel> \<currency>\<^sup>? A \<guillemotleft> suffix n) \<parallel> \<currency>\<^sup>+ A \<guillemotleft> suffix n"
```

Comment:
The problem I see is that the process families `𝟬 ∥ ¤⇧? A « suffix n) ∥ ¤⇧+ A « suffix n` and `𝟬 ∥ ¤⇧* A « suffix n` are not equal but bisimilar (since an application of `parallel_associativity` would be needed). Therefore, I think that, in line with what we did for `transition_from_duploss`, the statement of this lemma should be as follows:
```isabelle
lemma duploss_losing_transition:
  obtains Q where "¤⇧* A →⇩s⦇A ▹ ⋆⇗n⇖ X⦈ Q" and "Q ∼⇩s 𝟬 ∥ ¤⇧* A « suffix n"
```
What do you think?

Reply:
Yes, this would be the obvious choice. The question is whether this makes the proofs that use these lemmas more complicated. Are we using these lemmas somewhere at the moment, and, if yes, do we rely on them stating just a plain transition?

Reply:
I found two uses of these lemmas: in line 1254 and in line 1575. For the former, things might get more complex in a sense, because we would have to perform a simulation step to show that the follow-up transition we construct has a counterpart that starts at `Q`. The same applies in principle also to the latter, but there we have currently quite some follow-up reasoning with that complex term `?T`, which might get considerably simplified if we turned to the new versions of those lemmas.

What do you think? Should we make the change you proposed and see where it leads us? If in some particular situation we find that it makes matters more complicated, we could still unfold `duploss` or `bidirectional_bridge` on the spot and reason with the `_transition` lemmas for the underlying constructs.

Reply:
I think that makes much sense and we should go for it.

Reply:
Great! Please make the necessary changes then. I’d like to encourage you to use the reworked lemmas wherever somehow feasible instead of going for the unfolding solution. It’s more high-level and thus should make the proofs cleaner and more understandable in general.

Reply:
Done in 44280bc4cc6fc8a6d35cb9787f247b006ff3f4ee and afbd55cd555615e1792948e1b44c7f17556ea888, please verify.

Reply:
> ```isabelle-theory
> lemma duploss_losing_transition:
>   obtains Q where "¤⇧* A →⇩s⦇A ▹ ⋆⇗n⇖ X⦈ Q" and "Q ∼⇩s 𝟬 ∥ ¤⇧* A « suffix n"
> ```

Actually, I think the two statements after `where` should be swapped, because, as far as I remember, we typically first say what the obtained variables are about.

Reply:
Hmm, on the other hand, as it is now, the statement reads more naturally. Also, in contrast to, say, the `transition_from_` lemmas, this lemma doesn’t tell us anything about variables used in some assumption. Maybe we should leave the statement as written above.